### ISSUE TITLE
GPA module incorrectly truncates filenames if it contains more than one '.' in filename 

### ISSUE THREAD

**Issue by muratmaga** [SlicerMorph/SlicerMorph#344]
**Describe the bug**
An output from DeCaL is shown in the screenshot. The full name of the files contain .ply followed by _align_merged_mrk.json. When GPA is loading this type of file, it is truncating the filename to the first .ply. 

<img width="994" alt="image" src="https://github.com/user-attachments/assets/b0529927-48be-42ad-bb3d-e422fdb220b3">

The problem is when we try to import this dataset into R via SlicerMorphR using the analysis.json file, files cannot be located, because they are incorrectly truncated. See the first lines of analysis.json file:
```
{
  "@schema": "https://raw.githubusercontent.com/slicermorph/slicermorph/master/GPA/Resources/Schema/GPALog-schema-v1.0.0.json#",
  "GPALog": [
    {
      "Date": "2024-09-13",
      "Time": "16:39:14",
      "InputPath": "/Users/amaga/Downloads/merged",
      "OutputPath": "/Users/amaga/Downloads/merged/2024-09-13_16_39_13",
      "Files": [
        "USGS_505_ML344.mrk.json",
        "USGS_3844_ML348.mrk.json",
        "USGS_3860_ML364.mrk.json",
        "USGS_5937_ML180.mrk.json",
        "USGS_7197_ML180.mrk.json",
        "USGS_9202_ML344.mrk.json",
        "USGS_27426_ML315.mrk.json",
        "USNM_493928_ML212.mrk.json",
        "USNM_495378_ML360.mrk.json",
        "USNM_521653_ML212.mrk.json",
        "USNM_527507_ML212.mrk.json",
        "USNM_527633_ML290.mrk.json",
        "USNM_545035_ML290.mrk.json",
        "YPM_30503_ML336.mrk.json",
        "YPM_30504_ML336.mrk.json",
        "YPM_30704_ML190.mrk.json",
        "YPM_30707_ML190.mrk.json",
        "YPM_PU_17419.mrk.json"
      ],
      "LMFormat": ".mrk.json",
      "NumberLM": 314,
      "ExcludedLM": [],
      "Boas": false,
      "MeanShape": "meanShape.csv",
      "Eigenvalues": "eigenvalues.csv",
      "Eigenvectors": "eigenvectors.csv",
      "OutputData": "outputData.csv",
      "PCScores": "pcScores.csv",
      "SemiLandmarks": [
```

**Comment by smrolfe** [comment]
I can handle this case in the GPA module, but it seems like it might be better to fix this issue in the DeCA module instead? I would prefer not to allow an extension in the middle of the file name, especially when we have to handle .mrk.json and .json extensions.

**Comment by muratmaga** [comment]
I think we need to do both. Fix, the Deca to remove filename extension, and handle mulitple dots in filenames in GPA. It is possible (and I have seen it) that people use "." in the filename, not necessarily as an extension as a spacer.

**Comment by smrolfe** [comment]
Addressed in commit e70afd7b46b84d8869a8d5c529d2ccc7f16d12b2
Tested in the Preview version

**Comment by muratmaga** [comment]
verified working
